src/network/networkd-netdev-wireguard.c

```diff
@@ -0,0 +1,625 @@
+                end = strchr(rvalue, ']');
+                if (!end) {
+                        log_syntax(unit, LOG_ERR, filename, line, 0, "Unable to find matching brace of endpoint, ignoring assignment: %s", rvalue);
+                        return 0;
+                }
+                len = end - begin;
+                ++end;
+                if (*end != ':' || !*(end + 1)) {
+                        log_syntax(unit, LOG_ERR, filename, line, 0, "Unable to find port of endpoint: %s", rvalue);
+                        return 0;
+                }
+                ++end;
+        } else {
+                begin = rvalue;
+                end = strrchr(rvalue, ':');
+                if (!end || !*(end + 1)) {
+                        log_syntax(unit, LOG_ERR, filename, line, 0, "Unable to find port of endpoint: %s", rvalue);
+                        return 0;
+                }
+                len = end - begin;
+                ++end;
+        }
+
+        host = strndup(begin, len);
+        if (!host)
+                return -ENOMEM;
+
+        port = strdup(end);
+        if (!port)
+                return -ENOMEM;
+
+        endpoint->peer = peer;
+        endpoint->host = host;
+        endpoint->port = port;
+        LIST_PREPEND(endpoints, w->unresolved_endpoints, endpoint);
+        w->n_unresolved_endpoints++;
+
+        peer = NULL;
+        host = NULL;
+        port = NULL;
+        endpoint = NULL;
+
+        return 0;
+}
+
+int config_parse_wireguard_keepalive(const char *unit,
+                                     const char *filename,
+                                     unsigned line,
+                                     const char *section,
+                                     unsigned section_line,
+                                     const char *lvalue,
+                                     int ltype,
+                                     const char *rvalue,
+                                     void *data,
+                                     void *userdata) {
+        int r;
+        uint16_t keepalive = 0;
+        Wireguard *w;
+        WireguardPeer *peer;
+
+        assert(rvalue);
+        assert(data);
+
+        w = WIREGUARD(data);
+
+        assert(w);
+
+        peer = new_peer(w, section_line);
+        if (!peer)
+                return -ENOMEM;
+
+        // TODO implement "off" state?
+        r = safe_atou16(rvalue, &keepalive);
+        if (r < 0)
+                log_syntax(unit, LOG_ERR, filename, line, r, "The persistent keepalive interval must be 0-65535. Ignore assignment: %s", rvalue);
+
+        peer->fields.persistent_keepalive_interval = keepalive;
+        return 0;
+}
+
+static void wireguard_init(NetDev *netdev) {
+        Wireguard *w;
+
+        assert(netdev);
+
+        w = WIREGUARD(netdev);
+
+        assert(w);
+
+        w->allocation_size = sizeof(struct wgdevice);
+        w->dev.replace_peer_list = true;
+}
+
+int config_parse_wireguard_address_family(const char *unit,
```

Comment:
Since @Mic92 asked me about my opinion:

While that one can be handy switch I agree with @zx2c4. If you require an IPv4 or IPv6 only DNS record you can always add one (ip4.foo.bar / ip6.foo.bar). 

If you want to provide that option (besides DNS record) maybe default the address familiy to "auto" and then use the DNS result?